httpx/http_service.go

```diff
@@ -126,7 +149,8 @@ func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*htt
 		if resp.StatusCode >= 500 {
 			return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}
 		} else if resp.StatusCode >= 300 {
```

Comment:
I think you can just remove this else clause and let it fall through to the return below.